lib/routes/bing/daily-wallpaper.ts

```diff
@@ -1,41 +1,68 @@
 import { Route } from '@/types';
-import ofetch from '@/utils/ofetch';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
 import { parseDate } from '@/utils/parse-date';
 import timezone from '@/utils/timezone';
 
 export const route: Route = {
-    path: '/',
+    path: '/:routeParams?',
+    parameters: {
+        type: '返回的壁纸格式,取值为: UHD/1920x1080/1920x1200/768x1366/1080x1920/1080x1920_logo;默认值与输入的值不在范围内时设为1920x1080;1920x1200与1080x1920_logo会带有水印',
+        story: '是否开启壁纸对应故事, 取值为: 1/0;默认值为0即不开启故事输出',
+    },
     radar: [
         {
             source: ['cn.bing.com/'],
             target: '',
         },
     ],
     name: '每日壁纸',
-    maintainers: ['FHYunCai'],
+    maintainers: ['FHYunCai', 'LLLLLFish'],
     handler,
     url: 'cn.bing.com/',
+    example: '/bing/type=UHD&story=1',
 };
 
 async function handler(ctx) {
-    const response = await ofetch('HPImageArchive.aspx', {
-        baseURL: 'https://cn.bing.com',
-        query: {
-            format: 'js',
-            idx: 0,
-            n: ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 7,
-            mkt: 'zh-CN',
-        },
+    const routeParams = new URLSearchParams(ctx.req.param('routeParams'));
+    const lang = 'zh-CN';
+    let type = routeParams.get('type') || '1920x1080';
+    const allowedTypes = ['UHD', '1920x1080', '1920x1200', '768x1366', '1080x1920', '1080x1920_logo'];
+    if (!allowedTypes.includes(type)) {
+        type = '1920x1080';
+    }
+    const story = routeParams.get('story') === '1';
+    const api_url = `https://cn.bing.com/hp/api/model`;
+    const resp = await got({
+        method: 'get',
+        url: api_url,
     });
-    const data = response;
+    const items = await Promise.all(
+        resp.data.MediaContents.map((item) => {
+            const ssd = item.Ssd;
+            const key = `bing_${ssd}_${lang}_${type}_${story}`;
+            return cache.tryGet(key, () => {
+                const link = `https://cn.bing.com${item.ImageContent.Image.Url.match(/\/th\?id=[^_]+_[^_]+/)[0].replace(/(_\d+x\d+\.webp)$/i, '')}_${type}.jpg`;
```

Comment:
> Use WebP format instead of JPG whenever possible since it offers better compression.

docs: https://docs.rsshub.app/joinus/advanced/script-standard